Misc/stable_abi.txt

```diff
@@ -54,40 +63,58 @@
 
 struct PyObject
     added 3.2
+    fields ob_refcnt ob_type
 struct PyVarObject
     added 3.2
+    fields ob_base ob_size
 struct PyMethodDef
     added 3.2
+    full-abi
 struct PyMemberDef
     added 3.2
+    full-abi
 struct PyGetSetDef
     added 3.2
+    full-abi
 struct PyModuleDef_Base
     added 3.2
+    full-abi
 struct PyModuleDef
     added 3.2
+    full-abi
 struct PyStructSequence_Field
     added 3.2
+    full-abi
 struct PyStructSequence_Desc
     added 3.2
+    full-abi
 struct PyType_Slot
     added 3.2
+    full-abi
 struct PyType_Spec
     added 3.2
+    full-abi
 struct PyThreadState
     added 3.2
+    opaque
 struct PyInterpreterState
     added 3.2
+    opaque
 struct PyFrameObject
     added 3.2
+    opaque
```

Comment:
It seems like there is a misunderstanding on what "a structure" means here. Python C API is misleading because it uses short ``PyFrameObject`` type name rather than ``struct PyFrameObject``.

In practice in C, "PyFrameObject" is a type name, not a structure. I recently added ``Include/pytypedefs.h`` to clarify which type names are always available in the C API (and fix a compiler error), even if their related structures are opaque. I understand that the **type** (type "name") is part of the C API, but the structure is not.

I'm not sure how  https://docs.python.org/dev/c-api/frame.html should be rephrased. The follow this PR, maybe "The structure is opaque" which may become redundant once this PR is merged.

Reply:
Ah, I see.
This should be explained/avoided:  if you don't know the term "opaque structure" you are likely to search for it, but if you don't realize the distinction between "type" and "structure", you'll probably just be confused.

I'll work on the rephrasing next week. Please let me know if you know of other places with similar wording.